From ee973e7899a49634e9a9e5a508397da4e7c05265 Mon Sep 17 00:00:00 2001
From: "S. Gilles" <sgilles@math.umd.edu>
Date: Wed, 22 Feb 2017 13:04:53 +0000
Subject: [PATCH] Fix #3247

Since commit e87c039954 of github.com/libressl-portable/openbsd.git ,
LibreSSL has a few more X509_V_ERR_... values, and will complain if they
are redefined.
---
 src/_cffi_src/openssl/x509_vfy.py | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/_cffi_src/openssl/x509_vfy.py b/src/_cffi_src/openssl/x509_vfy.py
index 8bd6e41..979a618 100644
--- a/src/_cffi_src/openssl/x509_vfy.py
+++ b/src/_cffi_src/openssl/x509_vfy.py
@@ -221,10 +221,13 @@
 static const long X509_V_ERR_SUITE_B_INVALID_SIGNATURE_ALGORITHM = 0;
 static const long X509_V_ERR_SUITE_B_LOS_NOT_ALLOWED = 0;
 static const long X509_V_ERR_SUITE_B_CANNOT_SIGN_P_384_WITH_P_256 = 0;
+#if !defined(LIBRESSL_VERSION_NUMBER) || \
+    LIBRESSL_VERSION_NUMBER < 0x2050100fL
 static const long X509_V_ERR_HOSTNAME_MISMATCH = 0;
 static const long X509_V_ERR_EMAIL_MISMATCH = 0;
 static const long X509_V_ERR_IP_ADDRESS_MISMATCH = 0;
 #endif
+#endif
 
 /* OpenSSL 1.0.2beta2+ verification parameters */
 #if CRYPTOGRAPHY_OPENSSL_102BETA2_OR_GREATER && \
